iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java

```diff
@@ -517,6 +519,11 @@ public AmbientTypeReference AMBIENT_TYPE_REFERENCE() {
       f.ambientTypeReference(b.token(BicepLexicalGrammar.AMBIENT_TYPE_REFERENCE_VALUE)));
   }
 
+  public ArrayTypeReference ARRAY_TYPE_REFERENCE() {
+    return b.<ArrayTypeReference>nonterminal(BicepLexicalGrammar.ARRAY_TYPE_REFERENCE).is(
+      f.arrayTypeReference(b.token(BicepLexicalGrammar.ARRAY_TYPE_REFERENCE_VALUE)));
```

Comment:
By capturing the type expression as a single syntax token we lose some information (or will have to parse it later if we want the part without brackets).
I would suggest to structure it as in the grammar: extract `AMBIENT_TYPE_REFERENCE` and `IDENTIFIER` into `TYPE_REFERENCE`, add `TypeReference` interface with nullable fields for brackets and a token inside them.